src/averageAges.js

```diff
@@ -37,7 +39,14 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = withChildren
+    ? people.filter(person => person.sex === 'f'
+    && people.some(child => child.mother === person.name))
+    : people.filter(person => person.sex === 'f');
```

Comment:
also, you can make a check inside the callback, up to you
```suggestion
  const women = people.filter(person => withChildren
    ? person.sex === 'f'  && hasChildren(people, person.name)
    :  person.sex === 'f'
  );
```